stubs/jsonschema/jsonschema/validators.pyi

```diff
@@ -40,15 +40,27 @@ class _Validator:
     @classmethod
     def check_schema(cls, schema: Schema, format_checker: FormatChecker | Unset = ...) -> None: ...
     @property
+    @deprecated(
+        "Accessing resolver() is deprecated as of v4.18.0, "
+        "in favor of the https://github.com/python-jsonschema/referencing library."
+    )
     def resolver(self): ...
     def evolve(self, **changes) -> _Validator: ...
-    def iter_errors(self, instance, _schema: Schema | None = ...) -> Generator[Incomplete, None, None]: ...
+    @overload
+    def iter_errors(self, instance, _schema: None = None) -> Generator[Incomplete, None, None]: ...
+    @overload
+    @deprecated("Passing a schema to Validator.iter_errors is deprecated and will be removed in a future release.")
+    def iter_errors(self, instance, _schema: Schema | None = None) -> Generator[Incomplete, None, None]: ...
     def descend(
         self, instance, schema: Schema, path: Incomplete | None = ..., schema_path: Incomplete | None = ..., resolver=None
     ) -> Generator[Incomplete, None, None]: ...
     def validate(self, *args, **kwargs) -> None: ...
-    def is_type(self, instance, type): ...
-    def is_valid(self, instance, _schema: Schema | None = ...) -> bool: ...
+    def is_type(self, instance, type) -> bool: ...
+    @overload
+    def is_valid(self, instance, _schema: None = None) -> bool: ...
+    @overload
+    @deprecated("Passing a schema to Validator.is_valid is deprecated and will be removed in a future release.")
+    def is_valid(self, instance, _schema: Schema | None = None) -> bool: ...
```

Comment:
This is technically a bit more strict than the implementation, but matches the intent better:

```suggestion
    @overload
    def is_valid(self, instance) -> bool: ...
    @overload
    @deprecated("Passing a schema to Validator.is_valid is deprecated and will be removed in a future release.")
    def is_valid(self, instance, _schema: Schema | None) -> bool: ...
```